doc/sphinx/proof-engine/proof-handling.rst

```diff
@@ -539,6 +539,112 @@ Requesting information
+to it.  The third screen shot uses the "removed" option, so it shows the conclusion a
+second time with the old text, with deletions marked in red.  Also, since the hypotheses are
+new, no line of old text is shown for them.
+
+.. _diffs_coqtop_on:
+
+  .. image:: ../_static/diffs-coqtop-on.png
+     :alt: coqtop with Set Diffs on
+
+.. _diffs_coqide_on:
+
+  .. image:: ../_static/diffs-coqide-on.png
+     :alt: |CoqIDE| with Set Diffs on
+
+.. _diffs_coqide_removed:
+
+  .. image:: ../_static/diffs-coqide-removed.png
+     :alt: |CoqIDE| with Set Diffs removed
+
+**How to enable diffs**
+
+.. opt:: Set Diffs %( "on" %| "off" %| "removed" %)
+.. This ref doesn't work: :opt:`Set Diffs %( "on" %| "off" %| "removed" %)`
+
+The “on” option highlights added tokens in green, while the “removed” option
+additionally reprints items with removed tokens in red.  Unchanged tokens in
+modified items are shown with pale green or red.  (Colors are user-configurable.)
+
+For coqtop, showing diffs can be enabled when starting coqtop with the
+``-diffs on|off|removed`` command-line option or with the ``Set Diffs``
+command within Coq.  You will need to provide the ``-color on|auto`` command-line option when
+you start coqtop in either case.
+
+Colors for coqtop can be configured by setting the ``COQ_COLORS`` environment
+variable, which has the same format as the GNU ``LS_COLORS`` variable.  The string
+uses ANSI escape codes to represent attributes.  For example:
+
+        ``export COQ_COLORS=”diff.added=4;48;2;0;0;240:diff.added.bg=44”``
+
+sets the highlights for added text to underlined (the 4) with a background RGB
+color (0, 0, 240) and unmodified text in a modified item to a blue background.
+The other tags used by diffs are ``diff.removed`` and ``diff.removed.bg``.  Note
+that if you specify ``COQ_COLORS``, the predefined attributes are ignored.
+
+
+In CoqIDE, diffs should be enabled from the View menu.  Don’t use the ``Set Diffs``
+command in CoqIDE.  You can change the background colors shown for diffs from the
+Edit | Preferences | Tags panel by changing the settings for the ``diff.added``,
+``diff.added.bg``, ``diff.removed`` and ``diff.removed.bg`` tags.  This panel also
+allows let you control other attributes of the highlights, such as the foreground
+color, bold, italic, underline and strikeout.
+
+Note: As of this writing (August 2018), Proof General will need minor changes
+to be able to show diffs correctly.  We hope it will support this feature soon.
+
+**How diffs are calculated**
+
+Diffs are calculated as follows:
+
+1. Select the old proof state to compare to, which is the proof state before
+   the last tactic that changed the proof.  Changes that only affect the view
+   of the proof, such as ``all: swap 1 2``, are ignored.
+
+2. For each goal in the new proof state, determine what old goal to compare
+   it to--the one it is derived from or is the same as.  Match the hypotheses by
```

Comment:
did you mean to use — instead?

Reply:
Yes.  I expected Sphinx would convert it.  Apparently not, which awkward since there's no key on the keyboard for an em dash.

Reply:
> Does Windows offer something like compose keys?

I'm not aware of anything like that.  MS Word converts `--` to an em dash on the fly.  I haven't needed to type em dashes in other contexts in Windows.